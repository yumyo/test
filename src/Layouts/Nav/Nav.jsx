import React from 'react';
import PropTypes from 'prop-types';
import './Nav.scss';

const Nav = ({ children }) => [(
    <ul key="nav" className="nav-bar">
        <li><a href="/review">Part 1: Review</a></li>
        <li><a href="/task">Part 2: Task</a></li>
    </ul>
), children];

Nav.propTypes = {
    children: PropTypes.node.isRequired,
};

export default Nav;
