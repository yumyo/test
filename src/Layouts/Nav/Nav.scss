.nav-bar {
  background: #ebe1b7;

  list-style: none;
  padding: 0 20px;
  margin: 0;

  li {
    display: inline-block;
  }

  a {
    display: block;
    padding: 20px;
    color: #000;
    text-decoration: none;

    &:hover {
      background: #da9a5b;
    }
  }
}
