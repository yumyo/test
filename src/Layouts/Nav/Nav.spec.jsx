import React from 'react';
import { shallow } from 'enzyme';
import Nav from './Nav.jsx';

describe('<Nav />', () => {
    let component;

    beforeEach(() => {
        component = shallow(<Nav><p>a child</p></Nav>);
    });

    it('renders with children', () => {
        expect(component).toMatchSnapshot();
    });
});
