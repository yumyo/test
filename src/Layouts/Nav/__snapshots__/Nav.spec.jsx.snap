// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders with children 1`] = `
Array [
  <ul
    className="nav-bar"
    key="nav"
  >
    <li>
      <a
        href="/review"
      >
        Part 1: Review
      </a>
    </li>
    <li>
      <a
        href="/task"
      >
        Part 2: Task
      </a>
    </li>
  </ul>,
  <p>
    a child
  </p>,
]
`;
